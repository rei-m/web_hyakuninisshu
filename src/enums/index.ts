// このIFは苦肉の策だが他にいい方法はあるのか。。。
export interface EnumCollection<T> {
  readonly values: T[];
  readonly valueOf: (value: T) => T;
}

export enum MenuType {
  Training,
  Exam,
  Material,
  Other
}

export enum QuestionState {
  InAnswer,
  Answered,
  ConfirmCorrect,
  Finished
}

export enum RangeFromCondition {
  One = 1,
  Eleven = 11,
  TwentyOne = 21,
  ThirtyOne = 31,
  FortyOne = 41,
  FiftyOne = 51,
  SixtyOne = 61,
  SeventyOne = 71,
  EightyOne = 81,
  NinetyOne = 91
}

export const RangeFromConditions: EnumCollection<RangeFromCondition> = {
  valueOf: (value: number) => {
    const result = RangeFromConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeFrom, value is ${value}`);
  },
  values: [
    RangeFromCondition.One,
    RangeFromCondition.Eleven,
    RangeFromCondition.TwentyOne,
    RangeFromCondition.ThirtyOne,
    RangeFromCondition.FortyOne,
    RangeFromCondition.FiftyOne,
    RangeFromCondition.SixtyOne,
    RangeFromCondition.SeventyOne,
    RangeFromCondition.EightyOne,
    RangeFromCondition.NinetyOne
  ]
};

export enum RangeToCondition {
  Ten = 10,
  Twenty = 20,
  Thirty = 30,
  Forty = 40,
  Fifty = 50,
  Sixty = 60,
  Seventy = 70,
  Eighty = 80,
  Ninety = 90,
  OneHundred = 100
}

export const RangeToConditions: EnumCollection<RangeToCondition> = {
  valueOf: (value: number) => {
    const result = RangeToConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeTo, value is ${value}`);
  },
  values: [
    RangeToCondition.Ten,
    RangeToCondition.Twenty,
    RangeToCondition.Thirty,
    RangeToCondition.Forty,
    RangeToCondition.Fifty,
    RangeToCondition.Sixty,
    RangeToCondition.Seventy,
    RangeToCondition.Eighty,
    RangeToCondition.Ninety,
    RangeToCondition.OneHundred
  ]
};

export enum KimarijiCondition {
  None = 0,
  One = 1,
  Two = 2,
  Three = 3,
  Four = 4,
  Five = 5,
  Six = 6
}

export const KimarijiConditions: EnumCollection<KimarijiCondition> = {
  valueOf: (value: number) => {
    const result = KimarijiConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Kimariji, value is ${value}`);
  },
  values: [
    KimarijiCondition.None,
    KimarijiCondition.One,
    KimarijiCondition.Two,
    KimarijiCondition.Three,
    KimarijiCondition.Four,
    KimarijiCondition.Five,
    KimarijiCondition.Six
  ]
};

export enum ColorCondition {
  None = '',
  Blue = 'blue',
  Pink = 'pink',
  Yellow = 'yellow',
  Green = 'green',
  Orange = 'orange'
}

export const ColorConditions: EnumCollection<ColorCondition> = {
  valueOf: (value: string) => {
    const result = ColorConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Color, value is ${value}`);
  },
  values: [
    ColorCondition.None,
    ColorCondition.Blue,
    ColorCondition.Pink,
    ColorCondition.Yellow,
    ColorCondition.Green,
    ColorCondition.Orange
  ]
};

export enum KarutaStyleCondition {
  KanjiAndKana = 0,
  KanaOnly = 1
}

export const KarutaStyleConditions: EnumCollection<KarutaStyleCondition> = {
  valueOf: (value: number) => {
    const result = KarutaStyleConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown KarutaStyle, value is ${value}`);
  },
  values: [KarutaStyleCondition.KanjiAndKana, KarutaStyleCondition.KanaOnly]
};
