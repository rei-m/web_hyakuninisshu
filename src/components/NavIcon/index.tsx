import * as React from 'react';
import styled, { StyledFunction } from 'styled-components';
import { MenuType } from '@src/enums';
import { menuTypeToIcon } from '@src/utils';

export interface NavIconProps {
  readonly iconType: MenuType;
  readonly text: string;
  readonly isCurrent: boolean;
}

type IconProps = Pick<NavIconProps, 'text' | 'isCurrent'>;

const i: StyledFunction<IconProps & React.HTMLProps<HTMLElement>> = styled.i;

const Icon = i`
  height: 56px;
  color: #fff;
  text-align: center;
  padding-top: 6px;
  box-sizing: border-box;
  position: relative;
  width: inherit;
  opacity: ${props => (props.isCurrent ? 1 : 0.8)};
  &:after {
    content: '${props => props.text}';
    width: inherit;
    text-arign: center;
    position: absolute;
    bottom: 10px;
    left: 0;
    font-size: 1.2rem;
    opacity: 1;
  }
`;

const NavIcon: React.SFC<NavIconProps> = ({ iconType, text, isCurrent }) => (
  <Icon className="material-icons" text={text} isCurrent={isCurrent}>
    {menuTypeToIcon(iconType)}
  </Icon>
);

export default NavIcon;
