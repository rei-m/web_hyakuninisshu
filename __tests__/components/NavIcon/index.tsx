import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import NavIcon from '@src/components/NavIcon';
import { MenuType } from '@src/enums';

describe('<NavIcon />', () => {
  it('should render component when current', () => {
    const renderer = ReactTestRenderer.create(
      <NavIcon iconType={MenuType.Exam} text="hoge" isCurrent={true} />
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when not current', () => {
    const renderer = ReactTestRenderer.create(
      <NavIcon iconType={MenuType.Exam} text="hoge" isCurrent={false} />
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
