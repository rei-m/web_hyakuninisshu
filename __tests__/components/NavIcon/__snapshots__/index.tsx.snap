// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavIcon /> should render component when current 1`] = `
<span
  className="sc-bdVaJa iBGCyK"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1"
    focusable="false"
    role="presentation"
    style={
      Object {
        "fontSize": undefined,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M22 10l-6-6H4c-1.1 0-2 .9-2 2v12.01c0 1.1.9 1.99 2 1.99l16-.01c1.1 0 2-.89 2-1.99v-8zm-7-4.5l5.5 5.5H15V5.5z"
    />
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`<NavIcon /> should render component when not current 1`] = `
<span
  className="sc-bdVaJa yLiua"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1"
    focusable="false"
    role="presentation"
    style={
      Object {
        "fontSize": undefined,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M22 10l-6-6H4c-1.1 0-2 .9-2 2v12.01c0 1.1.9 1.99 2 1.99l16-.01c1.1 0 2-.89 2-1.99v-8zm-7-4.5l5.5 5.5H15V5.5z"
    />
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
  </svg>
</span>
`;
